src/turbolinks/browser_adapter.coffee

```diff
@@ -14,6 +14,7 @@ class Turbolinks.BrowserAdapter
   visitStarted: (visit) ->
     visit.issueRequest()
     visit.changeHistory()
+    visit.goToSamePageAnchor()
     visit.loadCachedSnapshot()
```

Comment:
I think I tested this and it doesn't seem to call `@adapter.visitRendered?(this)` twice. I will test again and confirm.